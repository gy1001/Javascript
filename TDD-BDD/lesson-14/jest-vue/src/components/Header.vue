<template>
  <div class="header">
    <div class="header-content">
      TodoList
      <input
        class="header-input"
        placeholder="请输入待办事项"
        type="text"
        data-test="input"
        v-model="inputValue"
        @keyup.enter="handlerAdd"
      />
    </div>
  </div>
</template>

<script>
export default {
  name: 'hello-world',
  props: {
    msg: String,
  },
  data() {
    return {
      inputValue: '',
    }
  },
  methods: {
    handlerAdd() {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    },
  },
}
</script>
<style scoped lang="stylus">
.header{
  background-color: #666;
  line-height: 60px;
  .header-content{
    width: 600px;
    margin: 0 auto;
    color: #fff;
    font-size: 24px;
    .header-input{
      float: right;
      width: 360px;
      margin-top :16px;
      height: 30px;
      line-height: 30px;
      padding-left: 10px;
    }
  }
}
</style>
