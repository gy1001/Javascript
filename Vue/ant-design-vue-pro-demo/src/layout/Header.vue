<template>
  <div class="header-container">header</div>
</template>

<script>
export default {};
</script>

<style lang="less" scoped>
.header-container{
  float: right;
}
</style>
