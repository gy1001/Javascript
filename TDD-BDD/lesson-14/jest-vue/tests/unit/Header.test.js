import { shallowMount } from '@vue/test-utils'
import Header from '@/components/Header.vue'

const wrapper = shallowMount(Header)
const input = wrapper.find('input[data-test="input"]')

describe('Header.vue', () => {
  it('Header 包含 input', () => {
    expect(input.exists()).toBe(true)
  })

  it('Header中 input的初始值为空', () => {
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('')
  })

  it('Header中 input输入内容时数据更新', () => {
    input.setValue('hello world')
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('hello world')
  })

  it('Header 中input 框输入回车时候，无内容时，无反应', () => {
    input.setValue('')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeFalsy()
  })

  it('Header 中input 框输入回车时候，有内容时，向外触发事件 add, 并清空 inputValue', () => {
    input.setValue('111')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
    expect(wrapper.vm.inputValue).toBe('')
  })

  it('header 样式发生改变，做提示', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
