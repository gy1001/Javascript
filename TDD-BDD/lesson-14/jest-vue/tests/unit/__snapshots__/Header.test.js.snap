// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue header 样式发生改变，做提示 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input placeholder="请输入待办事项" type="text" data-test="input" class="header-input">
  </div>
</div>
`;
